### DISCUSSION TITLE
nunjucks template

### DISCUSSION THREAD

**Discussion by GreatState** [storybookjs/storybook#16658] [Ideas]
**Is your feature request related to a problem? Please describe**
I would really like to use nunjucks with storybook. I'm struggling to install and get it set up

**Describe the solution you'd like**
I would like an option to add storybook/nunjucks to and existing repo

**Describe alternatives you've considered**
I've tried several nunjucks loaders, followed stack overflow issues and tried out many options but cannot get it working

**Are you able to assist to bring the feature to reality?**
no | yes, I can...

**Additional context**

previous 'closed' issue raised

https://github.com/storybookjs/storybook/issues/5506

I would love to get this working. I wish there was an official starter or option for this when installing. I'm using the latest version of storybook and cannot get this to work. simple nunjucks loader won't work for me and not sure if this is due to webpack versions

![image](https://user-images.githubusercontent.com/35596657/141086990-7f873a6b-56fa-44a3-b843-328d5e1f3407.png)

I've swapped this out for nunjucks-html-loader

```
use: [
        'html-loader',
        {
          loader: 'nunjucks-html-loader',
          options: {
            // Other super important. This will be the base
            // directory in which webpack is going to find
            // the layout and any other file index.njk is calling.
            searchPaths: [...returnEntries('./src/templates/**')],
          },
        },
      ],
```

but i get this error when it loads

![image](https://user-images.githubusercontent.com/35596657/141091636-4a18161b-71bf-432f-95f9-9dd3cc0c9ac5.png)

```
@babel/core": "^7.16.0",
    "@storybook/addon-actions": "^6.3.12",
    "@storybook/addon-essentials": "^6.3.12",
    "@storybook/addon-links": "^6.3.12",
    "@storybook/html": "^6.3.12",
    "nunjucks": "^3.2.2",
    "nunjucks-html-loader": "^1.1.0",
    "nunjucks-loader": "^3.0.0",

```

I'\ve tried using the nunjucks package inside the story

```
import { action } from '@storybook/addon-actions'
import Button from './button.njk'
import nunjucks from 'nunjucks'

export default {
  title: 'Design System/Atoms/Button',
  component: Button,
  argTypes: {
    label: { control: 'text' },
  },
}

export const HTMLTemplate = ({ className }) => {
  return nunjucks.render(Button({ className }))
}

const NunjucksTemplate = ({ className }) => {
  return 'nunjucks template coming soon'
}

export const Template = ({ onClick, label }) => {
  return Button({ label })
}

export const Text = Template.bind({})
Text.args = {
  label: 'Button',
}
```

**Comment by ogonkov** [comment]
Looks like you use `simple-nunjucks-loader` 3.x that works with webpack 5, with webpack 4. Try loader 2.x version
